### ISSUE TITLE
Regression: "entrypoint", "privileged" dev container feature settings no longer work

### ISSUE THREAD

**Issue by Chuxel** [devcontainers/cli#91]
This will break both docker-in-docker and docker-from-docker. 

Repro:
1. Put the following devcontainer.json on disk:
    ```json
    {
         "image": "debian",
         "features": {
             "docker-in-docker": "latest"
         }
     }
    ```
2. Install dev container CLI v0.8.0
3. `devcontainer up --workspace-folder .`
4. `devcontainer exec --workspace-folder . docker ps`

Expected: No error
Actual: `Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?`

There's an underlying permissions denied error here because `--privileged` isn't set like it is supposed to per the feature setup.  To see it, type: `devcontainer exec --workspace-folder . /usr/local/share/docker-init.sh`

You will see:
```
mount: /sys/kernel/security: permission denied.
Could not mount /sys/kernel/security.
AppArmor detection and --privileged mode might break.
mount: /tmp: permission denied.
```

This repros in the latest preview Remote - Containers and in the dev container Action and AzDO tasks that pulled in version `0.8.0`.

**Comment by Chuxel** [comment]
In fact, the entrypoint may no longer be getting fired either. :frown:

**Comment by Chuxel** [comment]
Appears to be fixed in 0.8.1
